### ISSUE TITLE
Redirection Error

### ISSUE THREAD

**Issue by mojoego** [johngodley/redirection#770]
```
Plugin: 3.2
WordPress: 4.9.4 (single)
PHP: 5.6.33
Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3343.3 Safari/537.36
JavaScript: https://shalumo.com/wp-content/plugins/redirection/redirection.js
REST API: https://shalumo.com/wp-json/

Error: Unexpected token < in JSON at position 0
Action: redirection/v1/redirect/ POST
Params: {"id":0,"url":"/samuel-berthomeau/","title":"","regex":false,"match_type":"url","action_type":"url","position":0,"group_id":2,"action_code":301,"action_data":{"url":"/shalumo/"}}
Code: 403 Forbidden
Raw: <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>403 Forbidden</title>
</head><body>
<h1>Forbidden</h1>
<p>You don't have permission to access /wp-json/redirection/v1/redirect/
on this server.<br />
</p>
</body></html>

```

**Comment by mojoego** [comment]
Comment : 

- all plugin disabled
- Permalink set as custom

**Comment by johngodley** [comment]
You have something blocking https://shalumo.com/wp-json/. Can you try changing the dropdown setting in the error and see if it works better?

**Comment by mojoego** [comment]
Same same :(
### Raw
```Plugin: 3.2
WordPress: 4.9.4 (single)
PHP: 5.6.33
Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3343.3 Safari/537.36
JavaScript: https://shalumo.com/wp-content/plugins/redirection/redirection.js
REST API: http://shalumo.com/index.php?rest_route=/

Error: Failed to fetch
```

### Proxy
```Plugin: 3.2
WordPress: 4.9.4 (single)
PHP: 5.6.33
Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3343.3 Safari/537.36
JavaScript: https://shalumo.com/wp-content/plugins/redirection/redirection.js
REST API: https://shalumo.com/wp-admin/admin-ajax.php?action=red_proxy&rest_path=

Error: Unexpected token < in JSON at position 0
Action: redirection/v1/redirect/ POST
Params: {"id":0,"url":"/samuel-berthomeau/","title":"","regex":false,"match_type":"url","action_type":"url","position":0,"group_id":1,"action_code":301,"action_data":{"url":"/shalumo/"}}
Code: 403 Forbidden
Raw: <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>403 Forbidden</title>
</head><body>
<h1>Forbidden</h1>
<p>You don't have permission to access /wp-admin/admin-ajax.php
on this server.<br />
</p>
</body></html>
```

### Form Request
```Plugin: 3.2
WordPress: 4.9.4 (single)
PHP: 5.6.33
Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3343.3 Safari/537.36
JavaScript: https://shalumo.com/wp-content/plugins/redirection/redirection.js
REST API: https://shalumo.com/wp-admin/tools.php?page=redirection.php&action=red_proxy&rest_path=

Error: Unexpected token < in JSON at position 0
Action: redirection/v1/redirect/ POST
Params: {"id":0,"url":"/samuel-berthomeau/","title":"","regex":false,"match_type":"url","action_type":"url","position":0,"group_id":1,"action_code":301,"action_data":{"url":"/shalumo/"}}
Code: 403 Forbidden
Raw: <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>403 Forbidden</title>
</head><body>
<h1>Forbidden</h1>
<p>You don't have permission to access /wp-admin/tools.php
on this server.<br />
</p>
</body></html>
```

**Comment by AmyWW1** [comment]
My jetpack won't load either. Interesting. I am far from a geek. Not sure what to do!

**Comment by johngodley** [comment]
Yeah, it's probably the same problem with Jetpack as with this. There's definitely something blocking access in some way. From the error messages you provided above I suspect it's something on your server, as opposed to WordPress itself

Do you have anything in your .htaccess file that looks like it might block things? Do you know if your server is behind a proxy of any kind like Varnish?

**Comment by mojoego** [comment]
Sorry for the late reply and thanks for your answer!
.htaccess ok (see below), that's what bothering me, cause everything seem alright now. That's a customer website which was hacked, every seems to be clean now but this problems is the last I can't treat...

```
RewriteEngine On 
RewriteCond %{SERVER_PORT} 80 
RewriteRule ^(.*)$ https://shalumo.com/$1 [R,L]

# BEGIN WordPress
<IfModule mod_rewrite.c>
RewriteEngine On
RewriteBase /
RewriteRule ^index\.php$ - [L]
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule . /index.php [L]
</IfModule>

# END WordPress```

**Comment by johngodley** [comment]
Sorry for the delay. I am seeing this error for a few people, but for different reasons.

In one instance Varnish was removing cookies, and so the request wasn't authenticated. In another I found the server was rejecting `application/json` requests for some reason.

I have a new version 3.2.1 which I think works around the application/json issue.

https://github.com/johngodley/redirection/releases/tag/3.2.1

Can you give it a try and see if it helps? If you still get the error you should try the other settings in the dropdown

**Comment by mojoego** [comment]
Same same (see error below).... but frankly I'm not sure it's your plugin related... I get the same JSON error wen I try to activate jetpack features... 
The website seems to be all clean after the hack but maybe the server or the install is not fully okay... and I don't want to spend more time investigating something that I can't find a workaround to...
But if you get some ideas or if I can be any usefull, don't hesitate to ask!

```Plugin: 3.2
WordPress: 4.9.4 (single)
PHP: 7.2.1
Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3346.8 Safari/537.36
JavaScript: https://shalumo.com/wp-content/plugins/redirection/redirection.js
REST API: https://shalumo.com/wp-json/

Error: Unexpected token < in JSON at position 0
Action: redirection/v1/redirect/ POST
Params: {"id":0,"url":"/restaurant/","title":"","regex":false,"match_type":"url","action_type":"url","position":0,"group_id":1,"action_code":301,"action_data":{"url":"/"}}
Code: 403 Forbidden
Raw: <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>403 Forbidden</title>
</head><body>
<h1>Forbidden</h1>
<p>You don't have permission to access /wp-json/redirection/v1/redirect/
on this server.<br />
</p>
</body></html>
```

**Comment by reeslo** [comment]
Same issue for me, but only on PROD, no 403 into local websites :/

**Comment by johngodley** [comment]
@reeslo that is a different issue and is caused by something caching your site. See https://redirection.me/support/problems/cloudflare/

@mojoego, if it was ok with you to give me access to the site, I'd be willing to investigate further - I'd like to try and do as much as I can in the plugin to resolve these kind of problems. No problem if it's not possible though.

**Comment by reeslo** [comment]
I think Its related tout lastest WordPress version.

I tried by deactivating all plugins should cause problems : Wordfence, wprocket.

**Comment by reeslo** [comment]
Also on prod I checked apache mod security without any log related to this issue.

All our websites on prod with 'redirection'  have this issue.

**Comment by sevcik-poctivaagentura** [comment]
I have same issue on localhost but also in production server:
Plugin: 3.2
WordPress: 4.9.2 (single)
PHP: 5.6.24
Browser: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3457.0 Safari/537.36
JavaScript: https://onisystem.local/wp-content/plugins/redirection/redirection.js
REST API: https://onisystem.local/wp-json/

Error: Unexpected token < in JSON at position 0
Action: redirection/v1/bulk/redirect/delete/?orderby=id POST
Params: {"items":"69"}
Code: 200 OK
Raw: <br />
<b>Deprecated</b>:  Automatically populating $HTTP_RAW_POST_DATA is deprecated and will be removed in a future version. To avoid this warning set 'always_populate_raw_post_data' to '-1' in php.ini and use the php://input stream instead. in <b>Unknown</b> on line <b>0</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>Unknown</b> on line <b>0</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1274</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api.php</b> on line <b>544</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api.php</b> on line <b>545</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api.php</b> on line <b>546</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api.php</b> on line <b>547</b><br />
{"items":[{"id":67,"url":"\\\/zavolame-vam2\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/kontakty\\\/$"},"match_type":"url","title":"","hits":52,"regex":true,"group_id":1,"position":66,"last_access":"2.8.2017","enabled":true},{"id":66,"url":"\\\/zadost-o-nabidku\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/kontakty\\\/$"},"match_type":"url","title":"","hits":57,"regex":true,"group_id":1,"position":65,"last_access":"6.8.2017","enabled":true},{"id":65,"url":"\\\/zavolame-vam\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/kontakty\\\/$"},"match_type":"url","title":"","hits":61,"regex":true,"group_id":1,"position":64,"last_access":"5.8.2017","enabled":true},{"id":64,"url":"\\\/vyzkousejte\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/kontakty\\\/$"},"match_type":"url","title":"","hits":77,"regex":true,"group_id":1,"position":63,"last_access":"6.8.2017","enabled":true},{"id":63,"url":"\\\/zaznam-trasy\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/osobni-vozidla\\\/osobni-auta\\\/osobni-auta-statistiky-a-prehledy\\\/$"},"match_type":"url","title":"","hits":61,"regex":true,"group_id":1,"position":62,"last_access":"6.8.2017","enabled":true},{"id":62,"url":"\\\/dochazkovy-system\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/osobni-vozidla\\\/osobni-auta\\\/osobni-auta-evidence-dochazky\\\/$"},"match_type":"url","title":"","hits":88,"regex":true,"group_id":1,"position":61,"last_access":"7.8.2017","enabled":true},{"id":61,"url":"\\\/funkcnost-v-zahranici\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/stavebni-stroje\\\/nakladni-auta\\\/nakladni-auta-mapa\\\/$"},"match_type":"url","title":"","hits":96,"regex":true,"group_id":1,"position":60,"last_access":"7.8.2017","enabled":true},{"id":60,"url":"\\\/napojeni-pult-centralni-ochrany\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/lesni-zemedelske-stroje\\\/vyvazeci-technika\\\/vyvazeci-technika-zabezpeceni-vozidel\\\/strezeni-vyvazeci-technika\\\/$"},"match_type":"url","title":"","hits":63,"regex":true,"group_id":1,"position":59,"last_access":"7.8.2017","enabled":true},{"id":59,"url":"\\\/radarove-dohledani\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/osobni-vozidla\\\/luxusni-auta\\\/luxusni-auta-radiove-dohledavani\\\/$"},"match_type":"url","title":"","hits":55,"regex":true,"group_id":1,"position":58,"last_access":"7.8.2017","enabled":true},{"id":58,"url":"\\\/sms-o-havarii\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/preprava-logistika\\\/autobusy\\\/autobusy-sms-email-notifikace\\\/$"},"match_type":"url","title":"","hits":58,"regex":true,"group_id":1,"position":57,"last_access":"7.8.2017","enabled":true},{"id":57,"url":"\\\/sms-o-kradezi\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/preprava-logistika\\\/autobusy\\\/autobusy-sms-email-notifikace\\\/$"},"match_type":"url","title":"","hits":53,"regex":true,"group_id":1,"position":56,"last_access":"6.8.2017","enabled":true},{"id":56,"url":"\\\/sms-o-odtahu\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/preprava-logistika\\\/autobusy\\\/autobusy-sms-email-notifikace\\\/$"},"match_type":"url","title":"","hits":59,"regex":true,"group_id":1,"position":55,"last_access":"6.8.2017","enabled":true},{"id":55,"url":"\\\/sms-o-havarii-e-call\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/preprava-logistika\\\/autobusy\\\/autobusy-sms-email-notifikace\\\/$"},"match_type":"url","title":"","hits":21,"regex":true,"group_id":1,"position":54,"last_access":"31.7.2017","enabled":true},{"id":54,"url":"\\\/aktivni-dohled-dispecinku\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/cenik-produktu-a-sluzeb\\\/$"},"match_type":"url","title":"","hits":71,"regex":true,"group_id":1,"position":53,"last_access":"1.8.2017","enabled":true},{"id":53,"url":"\\\/kniha-jizd-2\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/osobni-vozidla\\\/osobni-auta\\\/osobni-auta-kniha-jizd\\\/$"},"match_type":"url","title":"","hits":96,"regex":true,"group_id":1,"position":52,"last_access":"31.7.2017","enabled":true},{"id":52,"url":"\\\/planovani-rozvozu-a-svozu-zbozi\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/stavebni-stroje\\\/nakladni-auta\\\/nakladni-auta-mapa\\\/$"},"match_type":"url","title":"","hits":101,"regex":true,"group_id":1,"position":51,"last_access":"1.8.2017","enabled":true},{"id":51,"url":"\\\/idea-oni-system\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/$"},"match_type":"url","title":"","hits":37,"regex":true,"group_id":1,"position":50,"last_access":"6.8.2017","enabled":true},{"id":50,"url":"\\\/oni-ekonomik\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/preprava-logistika\\\/autobusy\\\/autobusy-ekonomika-provozu\\\/$"},"match_type":"url","title":"","hits":37,"regex":true,"group_id":1,"position":49,"last_access":"27.7.2017","enabled":true},{"id":49,"url":"\\\/volba-tarifu\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/cenik-produktu-a-sluzeb\\\/$"},"match_type":"url","title":"","hits":39,"regex":true,"group_id":1,"position":48,"last_access":"31.7.2017","enabled":true},{"id":48,"url":"\\\/kontrola-dodrzovani-rychlosti\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/osobni-vozidla\\\/osobni-auta\\\/osobni-auta-statistiky-a-prehledy\\\/$"},"match_type":"url","title":"","hits":56,"regex":true,"group_id":1,"position":47,"last_access":"26.7.2017","enabled":true},{"id":47,"url":"\\\/vykazy-o-provozu-vozidel\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/osobni-vozidla\\\/osobni-auta\\\/osobni-auta-statistiky-a-prehledy\\\/$"},"match_type":"url","title":"","hits":133,"regex":true,"group_id":1,"position":46,"last_access":"7.8.2017","enabled":true},{"id":46,"url":"\\\/zpracovani-cestovnich-prikazu\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/osobni-vozidla\\\/osobni-auta\\\/osobni-auta-kniha-jizd\\\/$"},"match_type":"url","title":"","hits":66,"regex":true,"group_id":1,"position":45,"last_access":"7.8.2017","enabled":true},{"id":45,"url":"\\\/sledovani-auta\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/cenik-produktu-a-sluzeb\\\/$"},"match_type":"url","title":"","hits":67,"regex":true,"group_id":1,"position":44,"last_access":"5.8.2017","enabled":true},{"id":44,"url":"\\\/prenosna-jednotka-do-zapalovace-plug-and-play\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/produkty-a-sluzby\\\/gps-radiove-jednotky\\\/$"},"match_type":"url","title":"","hits":53,"regex":true,"group_id":1,"position":43,"last_access":"4.8.2017","enabled":true},{"id":43,"url":"\\\/styl-jizdy-ridice\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/osobni-vozidla\\\/osobni-auta\\\/osobni-auta-styl-jizdy\\\/$"},"match_type":"url","title":"","hits":65,"regex":true,"group_id":1,"position":42,"last_access":"6.8.2017","enabled":true}],"total":67}

Error: Unexpected token < in JSON at position 0
Action: redirection/v1/redirect/ POST
Params: {"id":0,"url":"/wp-content/uploads/2017/01/ONI_katalog_2018_1_CZ_web-2.pdf-2.pdf","title":"","regex":false,"match_type":"url","action_type":"url","position":0,"group_id":2,"action_code":301,"action_data":{"url":"/wp-content/uploads/2017/01/ONI_katalog_2018_1_CZ_web-3.pdf-3.pdf"}}
Code: 200 OK
Raw: <br />
<b>Deprecated</b>:  Automatically populating $HTTP_RAW_POST_DATA is deprecated and will be removed in a future version. To avoid this warning set 'always_populate_raw_post_data' to '-1' in php.ini and use the php://input stream instead. in <b>Unknown</b> on line <b>0</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>Unknown</b> on line <b>0</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1274</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api\class-wp-rest-server.php</b> on line <b>1248</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api.php</b> on line <b>544</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api.php</b> on line <b>545</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api.php</b> on line <b>546</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent in <b>C:\Programovani\onisystem.cz\onisystem-cz-bitbucket\wp-includes\rest-api.php</b> on line <b>547</b><br />
{"items":[{"id":70,"url":"\/wp-content\/uploads\/2017\/01\/ONI_katalog_2018_1_CZ_web-2.pdf-2.pdf","action_code":301,"action_type":"url","action_data":{"url":"\/wp-content\/uploads\/2017\/01\/ONI_katalog_2018_1_CZ_web-3.pdf-3.pdf"},"match_type":"url","title":"","hits":0,"regex":false,"group_id":2,"position":0,"last_access":"-","enabled":true},{"id":67,"url":"\\\/zavolame-vam2\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/kontakty\\\/$"},"match_type":"url","title":"","hits":52,"regex":true,"group_id":1,"position":66,"last_access":"2.8.2017","enabled":true},{"id":66,"url":"\\\/zadost-o-nabidku\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/kontakty\\\/$"},"match_type":"url","title":"","hits":57,"regex":true,"group_id":1,"position":65,"last_access":"6.8.2017","enabled":true},{"id":65,"url":"\\\/zavolame-vam\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/kontakty\\\/$"},"match_type":"url","title":"","hits":61,"regex":true,"group_id":1,"position":64,"last_access":"5.8.2017","enabled":true},{"id":64,"url":"\\\/vyzkousejte\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/kontakty\\\/$"},"match_type":"url","title":"","hits":77,"regex":true,"group_id":1,"position":63,"last_access":"6.8.2017","enabled":true},{"id":63,"url":"\\\/zaznam-trasy\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/osobni-vozidla\\\/osobni-auta\\\/osobni-auta-statistiky-a-prehledy\\\/$"},"match_type":"url","title":"","hits":61,"regex":true,"group_id":1,"position":62,"last_access":"6.8.2017","enabled":true},{"id":62,"url":"\\\/dochazkovy-system\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/osobni-vozidla\\\/osobni-auta\\\/osobni-auta-evidence-dochazky\\\/$"},"match_type":"url","title":"","hits":88,"regex":true,"group_id":1,"position":61,"last_access":"7.8.2017","enabled":true},{"id":61,"url":"\\\/funkcnost-v-zahranici\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/stavebni-stroje\\\/nakladni-auta\\\/nakladni-auta-mapa\\\/$"},"match_type":"url","title":"","hits":96,"regex":true,"group_id":1,"position":60,"last_access":"7.8.2017","enabled":true},{"id":60,"url":"\\\/napojeni-pult-centralni-ochrany\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/lesni-zemedelske-stroje\\\/vyvazeci-technika\\\/vyvazeci-technika-zabezpeceni-vozidel\\\/strezeni-vyvazeci-technika\\\/$"},"match_type":"url","title":"","hits":63,"regex":true,"group_id":1,"position":59,"last_access":"7.8.2017","enabled":true},{"id":59,"url":"\\\/radarove-dohledani\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/osobni-vozidla\\\/luxusni-auta\\\/luxusni-auta-radiove-dohledavani\\\/$"},"match_type":"url","title":"","hits":55,"regex":true,"group_id":1,"position":58,"last_access":"7.8.2017","enabled":true},{"id":58,"url":"\\\/sms-o-havarii\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/preprava-logistika\\\/autobusy\\\/autobusy-sms-email-notifikace\\\/$"},"match_type":"url","title":"","hits":58,"regex":true,"group_id":1,"position":57,"last_access":"7.8.2017","enabled":true},{"id":57,"url":"\\\/sms-o-kradezi\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/preprava-logistika\\\/autobusy\\\/autobusy-sms-email-notifikace\\\/$"},"match_type":"url","title":"","hits":53,"regex":true,"group_id":1,"position":56,"last_access":"6.8.2017","enabled":true},{"id":56,"url":"\\\/sms-o-odtahu\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/preprava-logistika\\\/autobusy\\\/autobusy-sms-email-notifikace\\\/$"},"match_type":"url","title":"","hits":59,"regex":true,"group_id":1,"position":55,"last_access":"6.8.2017","enabled":true},{"id":55,"url":"\\\/sms-o-havarii-e-call\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/preprava-logistika\\\/autobusy\\\/autobusy-sms-email-notifikace\\\/$"},"match_type":"url","title":"","hits":21,"regex":true,"group_id":1,"position":54,"last_access":"31.7.2017","enabled":true},{"id":54,"url":"\\\/aktivni-dohled-dispecinku\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/cenik-produktu-a-sluzeb\\\/$"},"match_type":"url","title":"","hits":71,"regex":true,"group_id":1,"position":53,"last_access":"1.8.2017","enabled":true},{"id":53,"url":"\\\/kniha-jizd-2\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/osobni-vozidla\\\/osobni-auta\\\/osobni-auta-kniha-jizd\\\/$"},"match_type":"url","title":"","hits":96,"regex":true,"group_id":1,"position":52,"last_access":"31.7.2017","enabled":true},{"id":52,"url":"\\\/planovani-rozvozu-a-svozu-zbozi\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/stavebni-stroje\\\/nakladni-auta\\\/nakladni-auta-mapa\\\/$"},"match_type":"url","title":"","hits":101,"regex":true,"group_id":1,"position":51,"last_access":"1.8.2017","enabled":true},{"id":51,"url":"\\\/idea-oni-system\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/$"},"match_type":"url","title":"","hits":37,"regex":true,"group_id":1,"position":50,"last_access":"6.8.2017","enabled":true},{"id":50,"url":"\\\/oni-ekonomik\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/preprava-logistika\\\/autobusy\\\/autobusy-ekonomika-provozu\\\/$"},"match_type":"url","title":"","hits":37,"regex":true,"group_id":1,"position":49,"last_access":"27.7.2017","enabled":true},{"id":49,"url":"\\\/volba-tarifu\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/cenik-produktu-a-sluzeb\\\/$"},"match_type":"url","title":"","hits":39,"regex":true,"group_id":1,"position":48,"last_access":"31.7.2017","enabled":true},{"id":48,"url":"\\\/kontrola-dodrzovani-rychlosti\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/osobni-vozidla\\\/osobni-auta\\\/osobni-auta-statistiky-a-prehledy\\\/$"},"match_type":"url","title":"","hits":56,"regex":true,"group_id":1,"position":47,"last_access":"26.7.2017","enabled":true},{"id":47,"url":"\\\/vykazy-o-provozu-vozidel\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/osobni-vozidla\\\/osobni-auta\\\/osobni-auta-statistiky-a-prehledy\\\/$"},"match_type":"url","title":"","hits":133,"regex":true,"group_id":1,"position":46,"last_access":"7.8.2017","enabled":true},{"id":46,"url":"\\\/zpracovani-cestovnich-prikazu\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/oborova-reseni\\\/osobni-vozidla\\\/osobni-auta\\\/osobni-auta-kniha-jizd\\\/$"},"match_type":"url","title":"","hits":66,"regex":true,"group_id":1,"position":45,"last_access":"7.8.2017","enabled":true},{"id":45,"url":"\\\/sledovani-auta\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/cenik-produktu-a-sluzeb\\\/$"},"match_type":"url","title":"","hits":67,"regex":true,"group_id":1,"position":44,"last_access":"5.8.2017","enabled":true},{"id":44,"url":"\\\/prenosna-jednotka-do-zapalovace-plug-and-play\\\/?$","action_code":301,"action_type":"url","action_data":{"url":"\\\/produkty-a-sluzby\\\/gps-radiove-jednotky\\\/$"},"match_type":"url","title":"","hits":53,"regex":true,"group_id":1,"position":43,"last_access":"4.8.2017","enabled":true}],"total":68}

**Comment by johngodley** [comment]
I think this problem is fixed in the 3.3 beta:

https://github.com/johngodley/redirection/releases/tag/3.3-beta1

@sevcik-poctivaagentura your error is unrelated, and you should disable error output on a production site.
